lib/remote/jsonrpcconnection.cpp

```diff
@@ -110,6 +110,10 @@ void JsonRpcConnection::WriteOutgoingMessages(boost::asio::yield_context yc)
 		if (!queue.empty()) {
 			try {
 				for (auto& message : queue) {
+					if (m_ShuttingDown) {
+						break;
```

Comment:
`queue` can get quite big, can't it? And your connection could be relatively slow. Fast enough that messages make it through, but slow enough to notice it. Now we have the choice:

* Stop early, so that no message/TLS is half-written (this PR)
* Try to get something through, risking being interrupted in the middle of a message

Reply:
I'm not sure what this is trying to tell me. There's always the possibility that a connection dies somewhere in the middle of a message, so that can always happen. The one thing that would be strange would be to stop somewhere in the middle of a message but then send a TLS close notify (i.e. `boost::asio:ssl::stream::async_shutdown()` or `SSL_shutdown()`).